source/UF/SizeTruncationandConnectedness.lagda

```diff
@@ -0,0 +1,346 @@
+Ian Ray, 4th Febuary 2024.
+
+Modifications made by Ian Ray on 14 October 2024.
+
+Modifications made by Ian Ray on 7 January 2025.
+
+We develop some results that relate size, truncation and connectedness from
+the following paper:
+Christensen, J.D. (2024), Non-accessible localizations.
+Journal of Topology, 17: e12336.
+https://doi.org/10.1112/topo.12336
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K #-}
+
+open import UF.FunExt
+
+module UF.SizeTruncationandConnectedness
+        (fe : Fun-Ext)
+       where
+
+open import MLTT.Spartan hiding (_+_)
+open import Notation.CanonicalMap
+open import UF.ConnectedTypes fe
+open import UF.Embeddings
+open import UF.Equiv
+open import UF.EquivalenceExamples
+open import UF.PropTrunc
+open import UF.SmallnessProperties
+open import UF.Size
+open import UF.Subsingletons
+open import UF.Subsingletons-FunExt
+open import UF.Truncations fe
+open import UF.TruncationLevels
+open import UF.TruncatedTypes fe
+open import UF.Univalence
+
+module _
+        (te : general-truncations-exist)
+        (ua : Univalence)
+        (𝓥 : Universe)
+       where
+
+ private
+  pt : propositional-truncations-exist
+  pt = general-truncations-give-propositional-truncations te
+
+ open import UF.ImageAndSurjection pt
+
+\end{code}
+
+We begin by giving some definitions that Dan uses in his paper. We will use
+the universe parameter 𝓥 as our point of reference for 'smallness'.
+
+\begin{code}
+
+ _is_locally-small : (X : 𝓤 ̇) → (n : ℕ) → 𝓤 ⊔ (𝓥 ⁺) ̇
+ X is zero locally-small = X is 𝓥 small
+ X is (succ n) locally-small = (x x' : X) → (x ＝ x') is n locally-small
+
+ locally-small-is-prop : {X : 𝓤 ̇} {n : ℕ}
+                       → is-prop (X is n locally-small)
+ locally-small-is-prop {_} {X} {zero} = being-small-is-prop ua X 𝓥
+ locally-small-is-prop {_} {X} {succ n} =
+  Π₂-is-prop fe (λ x y → locally-small-is-prop)
+
+ locally-small-is-upper-closed : {X : 𝓤 ̇} {n : ℕ}
+                               → X is n locally-small
+                               → X is (succ n) locally-small
+ locally-small-is-upper-closed {_} {X} {zero} = small-implies-locally-small X 𝓥
+ locally-small-is-upper-closed {_} {X} {succ n} X-loc-small x x' =
+  locally-small-is-upper-closed (X-loc-small x x')
+
+\end{code}
+
+Local smallness is closed under sigma and equivalence for each n : ℕ.
+
+TODO: Add other closure properties and maybe move this to size file(?).
+
+\begin{code}
+
+ locally-small-closed-under-≃ : {X : 𝓤 ̇} {Y : 𝓦 ̇} {n : ℕ}
```

Comment:
I suggest `local-smallness-is-closed-under--≃`. In general, chose terminology which matches what people would say when speaking in English.